src/main/java/com/github/sdonkov/PrefixTree.java

```diff
@@ -61,21 +65,18 @@ public String toString() {
     public boolean contains(CharSequence word) {
         TraverseResult traverseResult = traverse(word);
         if (traverseResult.index == word.length() && traverseResult.node.isEndOfWord()) {
-            //todo
-            System.out.println("Tree contains this word - " + word);
+            logger.debug("Tree contains this word - " + word);
```

Comment:
This is really a `trace` statement (see [here](https://softwareengineering.stackexchange.com/questions/279690/why-does-the-trace-level-exist-and-when-should-i-use-it-rather-than-debug) for a discussion), but that's not a big deal. However, string concatenation is expensive and should be done only when it matters. Here, the string is always constructed while it is being used only when debug logging is turned on. Log statements must be guarded with [a check](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/src-html/org/apache/logging/log4j/Logger.html#line.2371) whether logging will be performed whenever they involve more than the invocation itself.
An alternative to the check is to use [parameterized messages](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/Logger.html#trace-java.lang.String-java.lang.Object-) (preferred) or [lambdas](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/Logger.html#trace-java.lang.String-org.apache.logging.log4j.util.Supplier...-).